### PR TITLE
Update STM32H5 HAL driver, fix some DMA bugs

### PR DESCRIPTION
**Pull request by multiplemonomials**
### Summary of changes 

This MR fixes some test failures that we have been seeing with STM32H5 processors on the spi-basic DMA abort test.  Unfortunately, it seems like this isn't a scenario that STMicro tests when releasing their HAL, because I found two different issues with it.

Issue number 1 was that, in one of the DMA callbacks, it was concluding that the channel didn't abort successfully because a certain register flag (`DMA_CCR_EN`) was still set after aborting the channel.  
Debugger screenshot

I'm not entirely 100% sure *why* that flag was still set, but it seems like checking it was in error, because in the latest release of the HAL, they removed the code that does the check!
Check removed diff

Unfortunately, updating the HAL introduced a different issue, where aborting the transfer would fail to correctly abort the Rx DMA channel.  After some time going back and forth, I was finally able to work out what was going on.  I filed a bug with STMicro [here](https://github.com/STMicroelectronics/stm32h5xx_hal_driver/issues/8).

This MR updates Mbed to use the latest release (commit 7e8c7d8e) of the STM32H5 HAL driver, plus the patch mentioned in the issue above.

#### Impact of changes 

- Aborting DMA SPI transfers should work now on STM32H5

### Documentation 

None

----------------------------------------------------------------------------------------------------------------
### Pull request type 

    [X] Patch update (Bug fix / Target update / Docs update / Test update / Refactor)
    [] Feature update (New feature / Functionality change / New API)
    [] Major update (Breaking change E.g. Return code change / API behaviour change)

----------------------------------------------------------------------------------------------------------------
### Test results 

    [] No Tests required for this change (E.g docs only update)
    [] Covered by existing mbed-os tests (Greentea or Unittest)
    [X] Tests / results supplied as part of this PR
    
New test results for STM32H5 pushed here: https://mbed-ce.github.io/mbed-ce-test-tools/tests/index.html

----------------------------------------------------------------------------------------------------------------

### CODE REVIEW

File: targets/TARGET_STM/TARGET_STM32H5/STM32Cube_FW/Note_add_new_STM32_family.md [hunk 1]
```diff
@@ -1,15 +1,12 @@
 # How to add new STM32 family
 1. 	In path mbed-os\targets\TARGET_STM\ add new family folder, in our my case TARGET_STM32H5, and also do not forget add same name into CmakeLists.txt in same folder level.
 2.	In path mbed-os\targets\TARGET_STM\TARGET_STM32H5\ prepare another folders. First is STM32Cube_FW and second CMSIS
-3. 	Visit https://github.com/STMicroelectronics. Select package for your family and chose latest tag. Then download it as a zip.
```
Comment (multiplemonomials) [inline targets/TARGET_STM/TARGET_STM32H5/STM32Cube_FW/Note_add_new_STM32_family.md:4]: These instructions sadly don't work anymore now that the HAL driver has moved to be a submodule (which is, on the balance, a good thing!).  I tried to update them as best I could.

File: targets/TARGET_STM/TARGET_STM32H5/STM32Cube_FW/stm32h5xx_hal_conf.h [hunk 2]
```diff
@@ -422,9 +422,12 @@ in voltage and temperature.*/
 #include "stm32h5xx_hal_i2s.h"
 #endif /* HAL_I2S_MODULE_ENABLED */
 
-#ifdef HAL_I3C_MODULE_ENABLED
-#include "stm32h5xx_hal_i3c.h"
-#endif /* HAL_I3C_MODULE_ENABLED */
+// Mbed patch to prevent circular include: ll_i3c.h includes this header, but hal_i3c.h needs ll_i3c.h to compile.
```
Comment (multiplemonomials) [inline targets/TARGET_STM/TARGET_STM32H5/STM32Cube_FW/stm32h5xx_hal_conf.h:425]: For the life of me, I could NOT get this new HAL release to compile due to this circular include issue.  I have NO IDEA how it ever worked.  Removing this include fixed the issue, and Mbed doesn't currently use i3c internally, so hopefully this is OK

File: targets/TARGET_STM/stm_spi_api.c [hunk 3]
```diff
@@ -1738,7 +1738,9 @@ static bool spi_master_start_asynch_transfer(spi_t *obj, transfer_type_t transfe
             LL_SPI_Enable(SPI_INST(obj));
         }
 #endif
-        DEBUG_PRINTF("SPI: RC=%u\n", rc);
+
+        // Unfortunately there is no way to propagate the error code back up, better to print a warning than swallow it entirely
+        printf("Warning: async SPI transfer start failed in STM32 HAL, error code = %d", rc);
```
Comment (multiplemonomials) [inline targets/TARGET_STM/stm_spi_api.c:1743]: Updated this HAL to print warnings if DMA operations fail.  These warnings would have saved me a lot of time debugging the code, as I didn't know that the error codes were being swallowed.

Comment (JohnK1987) [reply targets/TARGET_STM/stm_spi_api.c:1743]: What about commented out only the line 60? That will do same thing but how it was designed probably.

File: targets/TARGET_STM/TARGET_STM32H5/STM32Cube_FW/stm32h5xx_hal_conf.h (hunk 2 above)
Comment (JohnK1987) [reply targets/TARGET_STM/TARGET_STM32H5/STM32Cube_FW/stm32h5xx_hal_conf.h:425]: What about commented out only the line 60? That will do same thing but how it was designed probably.

Comment (multiplemonomials) [reply targets/TARGET_STM/TARGET_STM32H5/STM32Cube_FW/stm32h5xx_hal_conf.h]: OK I checked how STM32CubeIDE does it.  It looks like the IDE actually does not allow you to have both the HAL driver and the LL driver (the .c, not the .h) in the project at the same time.  That actually does solve the build error, so I will do that here.

**Comment by multiplemonomials** [comment]
OK this should be mergeable now, I added a new way in the license scanner to exempt the STM32Cube HAL directory from checking

**Review by JohnK1987 — APPROVED** [review]

**Pull request merged**